cloud/ec2_util.go

```diff
@@ -258,6 +258,22 @@ func expandUserData(userData string, expansions map[string]string) (string, erro
 	return expanded, nil
 }
 
+// 16kB
+const userDataSizeLimit = 16 * 1024
+
+func validateUserDataSize(userData, distroID string) error {
+	if len(userData) < userDataSizeLimit {
```

Comment:
wouldn't we want to check `ceil(n/3)*4`
(taken from [here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-add-user-data.html))

Reply:
True. I moved this line up so it checks before base64 encoding.

Reply:
still, potentially a user could use non-ASCII characters in their scripts and they'd be represented with more than one byte (in UTF-8)? I could hear ignoring this, though.

Reply:
This should be fine because [len()](https://golang.org/pkg/builtin/#len) for string is defined as the number of bytes, not the number of runes.